[2023-03-30T20:06:12.643-0400] {taskinstance.py:1084} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: air_quality_pipeline.run_producer scheduled__2023-03-30T00:00:00+00:00 [queued]>
[2023-03-30T20:06:12.659-0400] {taskinstance.py:1084} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: air_quality_pipeline.run_producer scheduled__2023-03-30T00:00:00+00:00 [queued]>
[2023-03-30T20:06:12.660-0400] {taskinstance.py:1282} INFO - 
--------------------------------------------------------------------------------
[2023-03-30T20:06:12.660-0400] {taskinstance.py:1283} INFO - Starting attempt 2 of 2
[2023-03-30T20:06:12.660-0400] {taskinstance.py:1284} INFO - 
--------------------------------------------------------------------------------
[2023-03-30T20:06:12.675-0400] {taskinstance.py:1303} INFO - Executing <Task(BashOperator): run_producer> on 2023-03-30 00:00:00+00:00
[2023-03-30T20:06:12.687-0400] {standard_task_runner.py:55} INFO - Started process 35324 to run task
[2023-03-30T20:06:12.694-0400] {standard_task_runner.py:82} INFO - Running: ['airflow', 'tasks', 'run', 'air_quality_pipeline', 'run_producer', 'scheduled__2023-03-30T00:00:00+00:00', '--job-id', '76', '--raw', '--subdir', 'DAGS_FOLDER/__pycache__/pipeline.py', '--cfg-path', '/var/folders/8z/pwbx6b3n3mnd1g1f397fv1100000gn/T/tmplyzg05ag']
[2023-03-30T20:06:12.696-0400] {standard_task_runner.py:83} INFO - Job 76: Subtask run_producer
[2023-03-30T20:06:12.755-0400] {task_command.py:388} INFO - Running <TaskInstance: air_quality_pipeline.run_producer scheduled__2023-03-30T00:00:00+00:00 [running]> on host srivardhans-air.starry-inc.net
[2023-03-30T20:06:12.826-0400] {taskinstance.py:1511} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=yourname
AIRFLOW_CTX_DAG_ID=air_quality_pipeline
AIRFLOW_CTX_TASK_ID=run_producer
AIRFLOW_CTX_EXECUTION_DATE=2023-03-30T00:00:00+00:00
AIRFLOW_CTX_TRY_NUMBER=2
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2023-03-30T00:00:00+00:00
[2023-03-30T20:06:12.828-0400] {subprocess.py:63} INFO - Tmp dir root location: 
 /var/folders/8z/pwbx6b3n3mnd1g1f397fv1100000gn/T
[2023-03-30T20:06:12.829-0400] {subprocess.py:75} INFO - Running command: ['/bin/bash', '-c', 'python3 "/Users/srivardhan/Desktop/Masters/Masters Project/Kafka Implementation/kafka-pipeline-examples/producer-consumer/producer.py"']
[2023-03-30T20:06:12.844-0400] {subprocess.py:86} INFO - Output:
[2023-03-30T20:07:13.629-0400] {subprocess.py:93} INFO - Fetching Data from Purple Air sensors
[2023-03-30T20:07:13.645-0400] {subprocess.py:93} INFO - Traceback (most recent call last):
[2023-03-30T20:07:13.646-0400] {subprocess.py:93} INFO -   File "/Users/srivardhan/Desktop/Masters/Masters Project/Kafka Implementation/kafka-pipeline-examples/producer-consumer/producer.py", line 23, in <module>
[2023-03-30T20:07:13.649-0400] {subprocess.py:93} INFO -     producer.send('air-quality', value=data)
[2023-03-30T20:07:13.650-0400] {subprocess.py:93} INFO -   File "/Users/srivardhan/opt/anaconda3/envs/kafka-pipeline/lib/python3.7/site-packages/kafka/producer/kafka.py", line 576, in send
[2023-03-30T20:07:13.652-0400] {subprocess.py:93} INFO -     self._wait_on_metadata(topic, self.config['max_block_ms'] / 1000.0)
[2023-03-30T20:07:13.653-0400] {subprocess.py:93} INFO -   File "/Users/srivardhan/opt/anaconda3/envs/kafka-pipeline/lib/python3.7/site-packages/kafka/producer/kafka.py", line 703, in _wait_on_metadata
[2023-03-30T20:07:13.654-0400] {subprocess.py:93} INFO -     "Failed to update metadata after %.1f secs." % (max_wait,))
[2023-03-30T20:07:13.654-0400] {subprocess.py:93} INFO - kafka.errors.KafkaTimeoutError: KafkaTimeoutError: Failed to update metadata after 60.0 secs.
[2023-03-30T20:07:13.704-0400] {subprocess.py:97} INFO - Command exited with return code 1
[2023-03-30T20:07:13.742-0400] {taskinstance.py:1775} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/Users/srivardhan/opt/anaconda3/envs/kafka-pipeline/lib/python3.7/site-packages/airflow/operators/bash.py", line 197, in execute
    f"Bash command failed. The command returned a non-zero exit code {result.exit_code}."
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2023-03-30T20:07:13.755-0400] {taskinstance.py:1326} INFO - Marking task as FAILED. dag_id=air_quality_pipeline, task_id=run_producer, execution_date=20230330T000000, start_date=20230331T000612, end_date=20230331T000713
[2023-03-30T20:07:13.784-0400] {standard_task_runner.py:105} ERROR - Failed to execute job 76 for task run_producer (Bash command failed. The command returned a non-zero exit code 1.; 35324)
[2023-03-30T20:07:13.838-0400] {local_task_job.py:212} INFO - Task exited with return code 1
[2023-03-30T20:07:13.901-0400] {taskinstance.py:2585} INFO - 0 downstream tasks scheduled from follow-on schedule check
